nomad/node_endpoint.go

```diff
@@ -245,10 +245,18 @@ func (n *Node) constructNodeServerInfoResponse(snap *state.StateSnapshot, reply
 	return nil
 }
 
-// Deregister is used to remove a client from the cluster. If a client should
-// just be made unavailable for scheduling, a status update is preferred.
+// Deregister is the deprecated single deregistration endpoint
 func (n *Node) Deregister(args *structs.NodeDeregisterRequest, reply *structs.NodeUpdateResponse) error {
-	if done, err := n.srv.forward("Node.Deregister", args, args, reply); done {
+	return n.BatchDeregister(&structs.NodeBatchDeregisterRequest{
```

Comment:
I don't think we can safely forward to `Node.BatchDeregister` here when servers are running mixed fleet.  Consider a 0.9.4 nomad server follower workin against 0.9.3 leader: the leader wouldn't be able to handle the call.

I think it's better to leave the old endpoints untouched and only add new ones.

Reply:
thanks, that makes a lot of sense